### PR TITLE
Document event subscriptions

### PR DESCRIPTION
**Pull request by aludvik**
This is a first pass "brain-dump" for how to subscribe to events. @chenette will be doing a second pass after this is merged to clean it up.

### CODE REVIEW

File: docs/source/app_developers_guide/event_subscriptions.rst [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+
+Requirements
+============
+
+Subscribing to Sawtooth events can be done in any language, provided the
+following are available.
+
+- A ZMQ library
+- A Protobuf library
+- The compiled Sawtooth protobuf messages for the chosen langauge
```
Comment (chenette) [inline docs/source/app_developers_guide/event_subscriptions.rst]: language

File: docs/source/app_developers_guide/event_subscriptions.rst [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+
+The following examples will be provided in Python, but the process should be
+similar for any imperative langauge.
```
Comment (chenette) [inline docs/source/app_developers_guide/event_subscriptions.rst]: language

File: docs/source/app_developers_guide/event_subscriptions.rst [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+
+
+Constructing a Subscription
+===========================
+
+Before submitting any messages to a validator, we must first specifiy the events
```
Comment (chenette) [inline docs/source/app_developers_guide/event_subscriptions.rst]: specify

File: docs/source/app_developers_guide/event_subscriptions.rst [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+    msg = Message(
+        correlation_id=correlation_id,
+        message_type=CLIENT_EVENTS_SUBSCRIBE_REQUEST,
+        content=request)
+
+    # Send the request
+    socket.send_multipart([msg.SerializeToString()])
+
+After sending the request, the validator will return a response indicating
+whether or not the subscription was succesful. The following example receives
```
Comment (chenette) [inline docs/source/app_developers_guide/event_subscriptions.rst]: successful

File: docs/source/app_developers_guide/event_subscriptions.rst [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+subscription when subscribing. If a block commit subscription is included, then
+all lists of events received from the validator, will contain a block commit
+event for the block the events came from.
+
+.. warning::
+
+    In forking networks, it is recommended that subscribers always include a
+    block commit subscription with the subscription request. This way the
+    subscriber can monitor for forks on the network and react appropriately.
+    Without a block commit subscription, there is not way to determine whether
```
Comment (chenette) [inline docs/source/app_developers_guide/event_subscriptions.rst]: not a way

File: docs/source/app_developers_guide/event_subscriptions.rst [hunk 6]
```diff
@@ -0,0 +1,216 @@
...
+    a fork has occured.
```
Comment (chenette) [inline docs/source/app_developers_guide/event_subscriptions.rst]: occurred

File: docs/source/app_developers_guide/event_subscriptions.rst [hunk 7]
```diff
@@ -0,0 +1,216 @@
...
+
+Event Catch-Up
+--------------
+
+When subscribing to events, it is also possible to request that all historical
+events since some known block be sent upon sucessful subscription. To use this
```
Comment (chenette) [inline docs/source/app_developers_guide/event_subscriptions.rst]: successful

**Comment by aludvik** [comment]
fixed typos

**Pull request merged**